include/ginkgo/core/matrix/batch_identity.hpp

```diff
@@ -0,0 +1,210 @@
+IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#ifndef GKO_PUBLIC_CORE_MATRIX_BATCH_IDENTITY_HPP_
+#define GKO_PUBLIC_CORE_MATRIX_BATCH_IDENTITY_HPP_
+
+
+#include <vector>
+
+
+#include <ginkgo/core/base/batch_lin_op.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/base/types.hpp>
+#include <ginkgo/core/base/utils.hpp>
+#include <ginkgo/core/matrix/identity.hpp>
+
+
+namespace gko {
+namespace matrix {
+
+
+/**
+ * BatchIdentity is a batch matrix format which explicitly stores all values of
+ * the matrix in each of the batches.
+ *
+ * The values in each of the batches are stored in row-major format (values
+ * belonging to the same row appear consecutive in the memory). Optionally, rows
+ * can be padded for better memory access.
+ *
+ * @tparam ValueType  precision of matrix elements
+ *
+ * @note While this format is not very useful for storing sparse matrices, it
+ *       is often suitable to store vectors, and sets of vectors.
+ * @ingroup batch_dense
+ * @ingroup mat_formats
+ * @ingroup BatchLinOp
+ */
+template <typename ValueType = default_precision>
+class BatchIdentity : public EnableBatchLinOp<BatchIdentity<ValueType>>,
+                      public EnableCreateMethod<BatchIdentity<ValueType>>,
+                      public BatchTransposable {
+    friend class EnableCreateMethod<BatchIdentity>;
+    friend class EnablePolymorphicObject<BatchIdentity, BatchLinOp>;
+
+public:
+    using EnableBatchLinOp<BatchIdentity>::convert_to;
+    using EnableBatchLinOp<BatchIdentity>::move_to;
+
+    using value_type = ValueType;
+    using index_type = int32;
+    using transposed_type = BatchIdentity<ValueType>;
+    using unbatch_type = Identity<ValueType>;
+    using mat_data = gko::matrix_data<ValueType, int64>;
+    using mat_data32 = gko::matrix_data<ValueType, int32>;
+    using absolute_type = remove_complex<BatchIdentity>;
+    using complex_type = to_complex<BatchIdentity>;
+
+    std::unique_ptr<BatchLinOp> transpose() const override
+    {
+        return this->clone();
+    }
+
+    std::unique_ptr<BatchLinOp> conj_transpose() const override
+    {
+        return this->clone();
+    }
+
+    /**
+     * Unbatches the batched dense and creates a std::vector of Identity
```

Comment:
I think this does not really unbatch the batch dense, but just creates a vector of Identity matrices from the given BatchIdentity matrix sizes

Reply:
Oh yes, I need to remove "dense" and replace it with "identity". I think there are many places in the docstrings that need fixing; I'll do that later.